File: app/src/test/kotlin/com/github/premnirmal/ticker/mock/MockTestRunner.java
package com.github.premnirmal.ticker.mock;

import android.app.Application;
import android.content.Context;
import android.support.test.runner.AndroidJUnitRunner;

/**
 * Created by premnirmal on 3/22/17.
 */
public final class MockTestRunner extends AndroidJUnitRunner {

  @Override
  public Application newApplication(ClassLoader cl, String className, Context context)
      throws InstantiationException, IllegalAccessException, ClassNotFoundException {
    return super.newApplication(cl, TestApplication.class.getName(), context);
  }
}

File: app/src/test/kotlin/com/github/premnirmal/ticker/mock/MockTestRunner.kt
package com.github.premnirmal.ticker.mock

import android.app.Application
import android.content.Context
import androidx.test.runner.AndroidJUnitRunner

/**
 * Created by premnirmal on 3/30/17.
 */
class MockTestRunner : AndroidJUnitRunner() {

  @Throws(InstantiationException::class, IllegalAccessException::class,
      ClassNotFoundException::class) override fun newApplication(cl: ClassLoader, className: String,
    context: Context): Application {
    return super.newApplication(cl, TestApplication::class.java.name, context)
  }
}